storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -809,32 +809,31 @@ func newPersistentVolume(
 		)
 	}
 
-	return &corev1.PersistentVolume{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:       volume.Name,
-			Labels:     map[string]string{},
-			Finalizers: []string{VOLUME_PROTECTION},
+	pv := corev1.PersistentVolume{
+		ObjectMeta: volume.Spec.Template.Metadata,
+		Spec:       volume.Spec.Template.Spec,
+	}
+	pv.ObjectMeta.Name = volume.Name
+	pv.ObjectMeta.Finalizers = []string{VOLUME_PROTECTION}
+	pv.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{
+		corev1.ReadWriteOnce,
+	}
+	pv.Spec.Capacity = map[corev1.ResourceName]resource.Quantity{
+		corev1.ResourceStorage: volumeSize,
+	}
+	pv.Spec.MountOptions = storageClass.MountOptions
```

Comment:
Overwrite or merge? Not sure.

Reply:
If you can specify the `MountOptions` directly in the PV, it kinds of defeat the purpose of `StorageClass` no?

I mean, one seeing a `Volume` with the `StorageClass` `Foo` will expect it to have the same mount options as the others `Volume` using that `StorageClass`.

I see it like `class` and `object` in languages.
I expect all the object instantiated from the class `Foo` to have the same interface. Now, sure, I can monkey-patch one specific instance but that's usually confusing and frowned upon (save for a few use case I guess).

Knowing it's always easier to ease a restrictions later than to reinforce it, I would go with `overwrite` now and if one day we have a good use case for the "monkey-patching" approach we can reconsider it, no?

Reply:
Agree. Over time we could maybe consider not enforcing a `StorageClass` in the `Volume`, only 'use' it if it's specified in the `PVTemplate`, and if none is specified, create a classless PV.

So, for now, all good, no need to change, and if we ever want to do the above change, will require a version bump.